.github/workflows/pr_cluster_workflow.yml

```diff
@@ -0,0 +1,115 @@
+# This workflow is designed for creating clusters using "config.yaml" file to implement GitOps solution with the help of "turnk8s", which includes 2 jobs.
+# It is possible to create or delete clusters based on the creating PRs, by updating the "config.yaml" file. In simple words, it describes the existing clusters on the server.
+# The "config.yaml" file can be updated in the following ways:
+#   1) For testing new features, the config.yaml file should only contain one cluster description(referred to as "test-cluster"), which must be in the "turnk8s-<PR_NUMBER>" format.This can be done by creating a draft PR.
+#   2) For modifications(creating, updating, or deleting clusters), simply create a normal PR and describe the cluster(s) within the "config.yaml" file. The clusters must not start with the "turnk8s-" prefix.
+#      *Note: Modifications take effect upon merging normal PRs into the main branch.
+#   3) All clusters are destroyed if the "config.yaml" file is empty.
+# The above-described cases are checked in the first job, called "checking_yaml_correctness".
+# The second job, "apply_cluster" starts only if the PR is a draft. As a result of the workflow, the cluster's kube-config file will be found attached in the "Artifacts" section.
+# *Note: The "test-cluster" is destroyed after merging the draft PR into the main branch.
+
+name: 'Automated Cluster Deployment for Pull Requests'
+
+on:
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'config.yaml'
+
+env:
+  TF_CLOUD_ORGANIZATION: "infraheads"
+  TF_API_TOKEN: "${{ secrets.TF_API_TOKEN }}"
+  TF_VAR_proxmox_token_id: "${{ secrets.PROXMOX_TOKEN_ID }}"
+  TF_VAR_proxmox_token_secret: "${{ secrets.PROXMOX_TOKEN_SECRET }}"
+  TF_VAR_github_token: "${{ secrets.TOKEN_GITHUB }}"
+  TF_VAR_netris_controller_host: "${{ vars.NETRIS_CONTROLLER_HOST }}"
+  TF_VAR_netris_controller_login: "${{ secrets.NETRIS_CONTROLLER_LOGIN }}"
+  TF_VAR_netris_controller_password: "${{ secrets.NETRIS_CONTROLLER_PASSWORD }}"
+  TF_VAR_argocd_admin_password: "${{ secrets.ARGOCD_ADMIN_PASSWORD }}"
+  TF_VAR_cluster_name: "turnk8s-${{ github.event.number }}"
+
+jobs:
+  checking_yaml_correctness:
+    runs-on: self-hosted
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+      
+      # * In case of empty config.yaml destroys all clusters.
+      # * In case of "turnk8s" logic changes, we conventionally create a draft PR and rely on the type of PR in the pipeline's logic which must be in the "turnk8s-<PR_NUMBER>" format.
+      # * For cluster creation we use normal PRs. In such PRs, the added cluster name should not have "turnk8s-" prefix.
+      - name: Ensure validity of the config.yaml file
+        run: |
+          if [ $(wc -c < "${GITHUB_WORKSPACE}/config.yaml") -ne 0  ];
```

Comment:
This does not cover hour cases, let's outline the cases.
1. The file size is 0; it is empty in any sense.
2. The file size is not 0. However, it contains only whiteface characters, which means from the human perspective, it is empty, but from the machine perspective, it is not. 
3. The file is not empty, but the yaml is malformed. (Should we handle this usecase or let yq error out?)
4. The file is not empty and properly structured.

Practical use cases.

1. All clusters have to be removed.
2. The file has changed, but no changes have been added, such as a new empty line to a non-empty file.
3. One cluster is added.
4. One cluster is removed.
5. One cluster is updated.
6. More than one clusters are added.
7. More than one clusters are removed.
8. More than one clusters are updated.
9. One cluster is added, and one cluster is removed.
10. Many clusters are added, one cluster is removed.
11. Many clusters are removed, one cluster is added.
12. 10 and one cluster is updated.
13. 11 and one cluster is updated.
14. Many clusters are added, and many clusters are removed.
15. Other cases.